Libraries/LibWeb/HTML/HTMLButtonElement.cpp

```diff
@@ -153,9 +155,108 @@ void HTMLButtonElement::activation_behavior(DOM::Event const& event)
             return;
     }
 
-    // FIXME: 4. Let target be the result of running element's get the commandfor associated element.
-    // FIXME: 5. If target is not null:
-    //           ...
+    // 4. Let target be the result of running element's get the commandfor associated element.
+    //    AD-HOC: Target needs to be an HTML Element in the following steps.
+    GC::Ptr<HTMLElement> target = as<HTMLElement>(m_command_for_element.ptr());
+    if (!target) {
+        auto target_id = attribute(AttributeNames::commandfor);
+        if (target_id.has_value()) {
+            root().for_each_in_inclusive_subtree_of_type<HTMLElement>([&](auto& candidate) {
+                if (candidate.attribute(HTML::AttributeNames::id) == target_id.value()) {
+                    target = &candidate;
+                    return TraversalDecision::Break;
+                }
+                return TraversalDecision::Continue;
+            });
+        }
+    }
+
+    // 5. If target is not null:
+    if (target) {
+        // 1. Let command be element's command attribute.
+        auto command = this->command();
+
+        // 2. If command is in the Unknown state, then return.
+        if (command.is_empty()) {
+            return;
+        }
+
+        // 3. Let isPopover be true if target's popover attribute is not in the no popover state; otherwise false.
+        auto isPopover = target->popover().has_value();
```

Comment:
This should be `snake_case`.

I believe this is equivalent to `target->has_attribute(AttributeNames::popover)` if you wanted to make this work for non HTML elements.